packages/@aws-cdk/aws-glue/lib/database.ts

```diff
@@ -86,13 +88,16 @@ export class Database extends Resource implements IDatabase {
    */
   public readonly locationUri?: string;
 
-  constructor(scope: Construct, id: string, props: DatabaseProps) {
+  constructor(scope: Construct, id: string, props: DatabaseProps = {}) {
     super(scope, id, {
-      physicalName: props.databaseName,
+      physicalName: props.databaseName ??
+        Lazy.string({
+          produce: () => Names.uniqueResourceName(this, {}).toLowerCase(),
+        }),
```

Comment:
> why use a `Lazy` here?

Because `physicalName` should set in the `super`. However, when we call the `super`, we can't use `this` for `Names` so we should use a `Lazy`. Also we can looks this pattern in other package (e.g `aws-stepfunctions/Activity`)
```ts
  constructor(scope: Construct, id: string, props: ActivityProps = {}) {
    super(scope, id, {
      physicalName: props.activityName ||
                Lazy.string({ produce: () => this.generateName() }),
    });
```
https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-stepfunctions/lib/activity.ts#L60-L64

> Do we have to use `toLowerCase()`?

Yes,  `AWS::Glue::Database`.`DatabaseInput` and `AWS::Glue::Table`.`TableInput.Name` must be lowercase.
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-glue-database-databaseinput.html
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-glue-table-tableinput.html

When I use uppercase, I got the error.
<img width="503" alt="image" src="https://user-images.githubusercontent.com/49480575/205921686-95bb7cd8-4501-4a83-b949-ffd0f07bf2f1.png">